helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -1288,6 +1308,19 @@ private void removeMessageFromZK(HelixDataAccessor accessor, Message message,
     }
   }
 
+  private void sendNopMessage(HelixDataAccessor accessor, String instanceName) {
+    try {
+      Message nopMsg = new Message(MessageType.NO_OP, UUID.randomUUID().toString());
+      nopMsg.setSrcName(instanceName);
+      nopMsg.setTgtName(instanceName);
+      accessor
+          .setProperty(accessor.keyBuilder().message(nopMsg.getTgtName(), nopMsg.getId()), nopMsg);
+      LOG.info("Send NO_OP message to " + nopMsg.getTgtName() + ", msgId: " + nopMsg.getId());
+    } catch (Exception e) {
+      LOG.error(e.toString());
```

Comment:
Can we have a more detailed error message here?

Reply:
I agree, it was copy-pasted. Let me make it complete.